### PR TITLE
Edit and delete lib implemented!

### PR DESCRIPTION
**Pull request by DanielleDauphinais**

### CODE REVIEW

**Comment by ejinks** [comment]
Looks great!

File: data/users.js [hunk 1]
```diff
@@ -79,93 +82,141 @@ let exportedMethods = {
...
-  
-    if (isValid) return { firstName, lastName, emailAddress: email, _id: _id.toString(), userName };
+
+    if (isValid)
+      return {
+        firstName,
+        lastName,
+        emailAddress: email,
+        _id: _id.toString(),
+        userName,
```
Comment (vishwa5854) [inline data/users.js:216]: Why are these changed ???

File: public/js/library/new.js [hunk 2]
```diff
@@ -1,122 +1,112 @@
...
+    errorList.hidden = true;
+    try {
+      validationsForStrings("Library Name", name.value, false, {
+        min: 3,
+        max: 40,
+      });
+      document.getElementById("nameError").innerText = "";
+    } catch (e) {
+      if (typeof e === "string" && e.startsWith("VError")) {
+        e = e.substr(1);
```
Comment (vishwa5854) [inline public/js/library/new.js:37]: What is changed here ?

Comment (vishwa5854) [reply public/js/library/new.js:37]: I mean the entire file

File: routes/libraries.js [hunk 3]
```diff
@@ -306,25 +524,153 @@ router
...
+
+    let id;
+    try {
+      id = req.params.id;
+      id = validation.checkValidId(id);
+    } catch (e) {
+      return res
+        .status(400)
+        .render("error", { errorCode: "400", searchValue: "Library ID" });
+    }
```
Comment (vishwa5854) [inline routes/libraries.js:609]: This is not how we are handling validation errors for 400s

File: routes/users.js [hunk 4]
```diff
@@ -53,24 +66,46 @@ router.post("/signup", async (req, res) => {
...
-        res.status(400).render('users/signup', { title: "Signup", error: `400 - ${err}` });
+    if (typeof err === "string")
+      return err.startsWith("VError")
+        ? res.status(400).render("users/signup", {
+            title: "Signup",
+            error: `400 - ${err.substr(1)}`,
+          })
+        : res
+            .status(400)
+            .render("users/signup", { title: "Signup", error: `400 - ${err}` });
```
Comment (vishwa5854) [inline routes/users.js:108]: This formatting is making the code hard to read
Why are we re-formatting these files ?

File: data/users.js (hunk 1 above)
Comment (DanielleDauphinais) [reply data/users.js:216]: Julien has the prettier formatter on so all these things got changed

File: public/js/library/new.js (hunk 2 above)
Comment (DanielleDauphinais) [reply public/js/library/new.js:37]: Julien uses the prettier formatter and that's what these changes are
